### PR TITLE
HTCONDOR-2389 submit-string-lists

### PR DESCRIPTION
**Pull request by JaimeFrey**
<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

**Review by GregThain — CHANGES_REQUESTED** [review]
One follow-one change requested.

File: src/condor_utils/submit_utils.cpp [hunk 1]
```diff
@@ -6163,32 +6140,28 @@ int SubmitHash::SetConcurrencyLimits()
...
-		StringList list(tmp.c_str());
+		std::vector<std::string> list = split(tmp);
 
-		char *limit;
-		list.rewind();
-		while ( (limit = list.next()) ) {
+		for (const auto& limit: list) {
 			double increment;
-			char *limit_cpy = strdup( limit );
+			char *limit_cpy = strdup(limit.c_str());
```
[the remark this answers is not in the document]
Comment (GregThain) [reply src/condor_utils/submit_utils.cpp:6150]: The "right" solution is to make `ParseConcurrencyLimits` take a (non-const) std::string, but that's a bunch of work to plumb out.

[the remark this answers is not in the document]
Comment (JaimeFrey) [reply src/condor_utils/submit_utils.cpp:6150]: That's tougher than it seems. `ParseConcurrencyLimit()` takes a `char*&` for some odd reason.

**Review by GregThain — APPROVED** [review]
LGTM.

**Pull request merged**